crates/swc_ecma_transforms_module/src/common_js.rs

```diff
@@ -55,6 +58,57 @@ where
     }
 }
 
+struct LazyIdentifierVisitor {
+    scope: Rc<RefCell<Scope>>,
+    top_level_idents: AHashSet<JsWord>,
+}
+impl LazyIdentifierVisitor {
+    fn new(scope: Rc<RefCell<Scope>>) -> Self {
+        LazyIdentifierVisitor {
+            scope,
+            top_level_idents: Default::default(),
+        }
+    }
+}
+
+impl Visit for LazyIdentifierVisitor {
+    noop_visit_type!();
+
+    fn visit_import_decl(&mut self, _: &ImportDecl) {}
+    fn visit_export_decl(&mut self, _: &ExportDecl) {}
+    fn visit_named_export(&mut self, _: &NamedExport) {}
+    fn visit_export_default_decl(&mut self, _: &ExportDefaultDecl) {}
+    fn visit_export_default_expr(&mut self, _: &ExportDefaultExpr) {}
+
+    fn visit_export_all(&mut self, export: &ExportAll) {
+        self.top_level_idents.insert(export.src.value.clone());
+    }
+
+    fn visit_function(&mut self, _: &Function) {}
```

Comment:
I think `visit_arrow_expr` is missing?

Reply:
Entirely possible! To be honest I don't really know JavaScript all that well (or, as you may have guessed, Rust for that matter). I tried to emulate the `mark_as_nested!` macro.

Can you give me an example of what would make an `ArrowExpr`? I'd like to add a test for it

Reply:
I didn't think a lot about it. I found several visitor methods to prevent visiting nested nodes, but I failed to find arrow expression from the method list.